pkg/controllers/disruption/consolidation_test.go

```diff
@@ -3508,58 +3364,492 @@ var _ = Describe("Consolidation", func() {
+				// bind pods to nodes
+				ExpectManualBinding(ctx, env.Client, pods[0], node1)
+				ExpectManualBinding(ctx, env.Client, pods[1], node2)
+				ExpectManualBinding(ctx, env.Client, pods[2], node3)
 
-			// wait for the controller to block on the validation timeout for single nodeclaim consolidation
-			Eventually(fakeClock.HasWaiters, time.Second*5).Should(BeTrue())
-			// advance the clock so that the timeout expires for single nodeclaim consolidation
-			fakeClock.Step(31 * time.Second)
+				// inform cluster state about nodes and nodeclaims
+				ExpectMakeNodesAndNodeClaimsInitializedAndStateUpdated(ctx, env.Client, nodeStateController, nodeClaimStateController, []*v1.Node{node1, node2, node3}, []*v1beta1.NodeClaim{nodeClaim1, nodeClaim2, nodeClaim3})
 
-			// controller should finish
-			Eventually(finished.Load, 10*time.Second).Should(BeTrue())
-			wg.Wait()
+				fakeClock.Step(10 * time.Minute)
 
-			// Process the item so that the nodes can be deleted.
-			ExpectReconcileSucceeded(ctx, queue, types.NamespacedName{})
+				var wg sync.WaitGroup
+				ExpectTriggerVerifyAction(&wg)
+				ExpectMakeNewNodeClaimsReady(ctx, env.Client, &wg, cluster, cloudProvider, 1)
+				ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKey{})
+				wg.Wait()
 
-			// Cascade any deletion of the nodeclaim to the node
-			ExpectNodeClaimsCascadeDeletion(ctx, env.Client, nodeClaim1, nodeClaim2, nodeClaim3)
+				// Process the item so that the nodes can be deleted.
+				ExpectReconcileSucceeded(ctx, queue, types.NamespacedName{})
 
-			// should have 2 nodes after single nodeclaim consolidation deletes one
-			Expect(ExpectNodeClaims(ctx, env.Client)).To(HaveLen(2))
-			Expect(ExpectNodes(ctx, env.Client)).To(HaveLen(2))
-			// and delete node3 in single nodeclaim consolidation
-			ExpectNotFound(ctx, env.Client, nodeClaim3, node3)
-		})
+				// Cascade any deletion of the nodeclaim to the node
+				ExpectNodeClaimsCascadeDeletion(ctx, env.Client, nodeClaim1, nodeClaim2, nodeClaim3)
+
+				// three nodeclaims should be replaced with a single nodeclaim
+				Expect(ExpectNodeClaims(ctx, env.Client)).To(HaveLen(1))
+				Expect(ExpectNodes(ctx, env.Client)).To(HaveLen(1))
+				ExpectNotFound(ctx, env.Client, nodeClaim1, node1, nodeClaim2, node2, nodeClaim3, node3)
+			},
+			Entry("if the candidate is on-demand node", false),
+			Entry("if the candidate is spot node", true),
+		)
+		DescribeTable("won't merge 2 nodes into 1 of the same type",
+			func(spotToSpot bool) {
+				leastExpInstance := leastExpensiveInstance
+				leastExpOffering := leastExpensiveOffering
+				nodeClaim1 := nodeClaims[0]
+				nodeClaim2 := nodeClaims[1]
+				node1 := nodes[0]
+				node2 := nodes[1]
+
+				if spotToSpot {
+					nodeClaim1 = spotNodeClaims[0]
+					nodeClaim2 = spotNodeClaims[1]
+					node1 = spotNodes[0]
+					node2 = spotNodes[1]
+					leastExpInstance = leastExpensiveSpotInstance
+					leastExpOffering = leastExpensiveSpotOffering
```

Comment:
I think you could continue to do ternaries here, I think it would reduce on the amount of code one has to reason about. 

```suggestion
				
				leastExpInstance := lo.Ternary(isSpotToSpot, leastExpensiveInstance, leastExpensiveSpotInstance)
				leastExpOffering := lo.Ternary(isSpotToSpot, leastExpensiveOffering, leastExpensiveSpotOffering)				
				nodeClaims = lo.Ternary(isSpotToSpot, nodeClaims, spotNodeClaims)
				nodes = lo.Ternary(isSpotToSpot, nodes, spotNodes)
```

Reply:
Addressed this.